graphql_api/types/user/user.py

```diff
@@ -44,4 +44,6 @@ def resolve_student_updated_at(user: Owner, info) -> Optional[datetime]:
 @user_bindable.field("customerIntent")
 def resolve_customer_intent(user: Owner, info) -> str:
     owner = user
+    if not owner.user:
```

Comment:
curious, why the misdirect `owner = user` above? Seems unnecessary

Reply:
I'm guessing because he want to showcase that the user is of type owner and we're fetching current from there?

@rohitvinnakota-codecov you can also do:
```
def resolve_customer_intent(current_owner: Owner, info) -> str:
    return current_owner.user.customer_intent
```

Reply:
Yes, this is mostly for readability. Otherwise it would read `user.user...`.

We have an issue to improve [this](https://github.com/codecov/engineering-team/issues/1220) so hopefully things will be more clear in the future.

Reply:
Got it, thanks!